pkg_resources/__init__.py

```diff
@@ -2008,46 +2014,55 @@ def find_on_path(importer, path_item, only=False):
     """Yield distributions accessible on a sys.path directory"""
     path_item = _normalize_cached(path_item)
 
-    if os.path.isdir(path_item) and os.access(path_item, os.R_OK):
-        if _is_unpacked_egg(path_item):
-            yield Distribution.from_filename(
-                path_item, metadata=PathMetadata(
-                    path_item, os.path.join(path_item, 'EGG-INFO')
-                )
+    if _is_unpacked_egg(path_item):
```

Comment:
If the code heads down this path now, but path_item is not a dir or is not readable, you'll get a different result than before. That seems to be a backward-incompatible change. Is this code path not expected to be impacted by this change?

Reply:
You're right, it probably needs some more work to, I'll dig into it.

Reply:
Actually… the code check that `path_item/EGG-INFO/PKG-INFO` is a file, so if that's the case, `path_item` has to be a directory. So the check seems useless.

The permission is not that useful either: the fact that the directory is readable is only useful when calling `os.listdir`, but no code in that path uses `os.listdir` – and if it does, it'd needs to learn to handle errors. What is interesting is to be sure the directory has `+x` permission, and that is done by using `os.path.isfile` in `_is_unpacked_egg`: it'll return `False` if it can't traverse the directory.

So… I can't see of a case where things would go wrong. If there's any we should definitely add a test!